cognite/client/utils/_token_generator.py

```diff
@@ -0,0 +1,57 @@
+import datetime
+import os
+
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+from cognite.client.exceptions import CogniteAPIKeyError
+
+
+class TokenGenerator:
+    def __init__(self):
+        if not self.environment_vars_set():
+            raise CogniteAPIKeyError("Missing required environment vars for token creation")
+
+        self._idp_client_id = os.getenv("COGNITE_IDP_CLIENT_ID")
+        self._idp_client_secret = os.getenv("COGNITE_IDP_CLIENT_SECRET")
+        self._idp_token_endpoint = os.getenv("COGNITE_IDP_TOKEN_ENDPOINT")
+
+        cognite_base_url = os.getenv("COGNITE_BASE_URL", "https://api.cognitedata.com")
+        self._idp_scope = [os.getenv("COGNITE_IDP_SCOPE", f"{cognite_base_url}/.default")]
```

Comment:
```suggestion
```suggestion
        self._idp_client_id = os.getenv("CLIENT_ID")
        self._idp_client_secret = os.getenv("CLIENT_SECRET")
        self._idp_token_endpoint = os.getenv("TOKEN_URL")

        cognite_base_url = os.getenv("COGNITE_BASE_URL", "https://api.cognitedata.com")
        self._idp_scope = [os.getenv("SCOPE", f"{cognite_base_url}/.default")]
```

The default scope works for Azure AD, but is likely to be different for other IdPs..

Reply:
If we think scope is likely to just not have a sensible default, I can make it a required parameter.